modules/programs/sftpman.nix

```
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.sftpman;
+
+  jsonFormat = pkgs.formats.json { };
+
+  mountOpts = { config, name, ... }: {
```

This would be well-suited as a [freeform module](https://nixos.org/manual/nixos/stable/#sec-freeform-modules):

```suggestion
  mountOpts = types.submodule ({ config, name, ... }: {
    freeformType = jsonFormat.type;
```

---

The JSON config format is unfortunately not documented, users are supposed to configure the mounts via the cli. The JSON keys are similar to the respective cli flags, but not exactly the same. So I thought it would be better to declare all the options explicitly.

---

Fair enough.